app/components/primer/dropdown.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Primer
+  # `Dropdown` is a lightweight context menu for housing navigation and actions.
+  # They're great for instances where you don't need the full power (and code) of the SelectMenu.
+  class Dropdown < Primer::Component
+    # Required trigger for the dropdown. Has the same arguments as <%= link_to_component(Primer::ButtonComponent) %>,
+    # but it is locked as a `summary` tag.
+    renders_one :button, lambda { |**system_arguments, &block|
+      @button_arguments = system_arguments
+      @button_arguments[:button] = true
+
+      view_context.capture { block&.call }
+    }
+
+    # Required context menu for the dropdown.
+    #
+    # @param as [Symbol] When `as` is `:list`, wraps the menu in a `<ul>` with a `<li>` for each item.
+    # @param direction [Symbol] <%= one_of(Primer::Dropdown::Menu::DIRECTION_OPTIONS) %>
+    # @param scheme [Symbol] Pass `:dark` for dark mode theming
+    # @param header [String] Optional string to display as the header
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
```

Comment:
Does it feel off to you that we write these comments here and not for the initializer on the slot class?

Reply:
it's just how we structure our docs. Adding the comments here will add a `### Slots` section and show the `menu` slot args in the same page as `Dropdown`.
We could restructure how we build it though

Reply:
👍🏻 that's what I thought. I think this is fine for now! Was just curious.